dm/registry.go

```diff
@@ -25,14 +25,25 @@ func init() {
 	r.MustRegisterWorkerType(WorkerDMSync, syncFactory)
 }
 
-type workerConstructor func(config lib.WorkerConfig) lib.Worker
+type workerConstructor func(lib.BaseWorker, lib.WorkerConfig) lib.WorkerImpl
```

Comment:
Should same as https://github.com/hanfei1991/microcosm/blob/149c0407683c8fc5c61c556434b6c95cd6ac2c8c/lib/registry/factory.go#L26 ?

Reply:
I want to wrap build BaseWorker, not call it three times for three units

Reply:
Master create worker by this interface https://github.com/hanfei1991/microcosm/blob/149c0407683c8fc5c61c556434b6c95cd6ac2c8c/executor/server.go#L159

Reply:
jobmaster create worker by this interface https://github.com/hanfei1991/microcosm/blob/149c0407683c8fc5c61c556434b6c95cd6ac2c8c/executor/server.go#L159
How we create a dump-Worker now?

Reply:
that is https://github.com/hanfei1991/microcosm/blob/149c0407683c8fc5c61c556434b6c95cd6ac2c8c/lib/registry/registry.go#L57 , in registryImpl.CreateWorker our factory (unitWorkerFactory) is found, then unitWorkerFactory.NewWorker, then the `type workerConstructor func(lib.BaseWorker, lib.WorkerConfig) lib.WorkerImpl`